docs/user-guide/configuring.md

```diff
@@ -382,6 +382,16 @@ The second way to use configuration files is via `.eslintrc` and `package.json`
 
 In each case, the settings in the configuration file override default settings.
 
+## Configuration File Formats
+
+ESLint supports configuration files in several formats:
+
+* **JavaScript** - use `.eslintrc.js` and export an object containing your configuration.
+* **YAML** - use `.eslintrc.yaml` or `.eslintrc.yml` to define the configuration structure.
+* **JSON** - use `.eslintrc.json` to define the configuration structure. ESLint's JSON files also allow JavaScript-style comments.
+* **package.json** - create an `eslintConfig` property in your `package.json` file and define your configuration there.
+* **Deprecated** - use `.eslintrc`, which can be either JSON or YAML.
```

Comment:
Just wondering, should we by favoring (or at least allowing) file names
without a leading dot, when they have extensions I mean?
On Nov 14, 2015 10:10 AM, "Nicholas C. Zakas" <notifications@github.com>
wrote:

> In docs/user-guide/configuring.md
> <https://github.com/eslint/eslint/pull/4406#discussion_r44860133>:
>
> > @@ -382,6 +382,16 @@ The second way to use configuration files is via `.eslintrc` and `package.json`
> >
> >  In each case, the settings in the configuration file override default settings.
> >
> > +## Configuration File Formats
> > +
> > +ESLint supports configuration files in several formats:
> > +
> > +* **JavaScript** - use `.eslintrc.js` and export an object containing your configuration.
> > +* **YAML** - use `.eslintrc.yaml` or `.eslintrc.yml` to define the configuration structure.
> > +* **JSON** - use `.eslintrc.json` to define the configuration structure. ESLint's JSON files also allow JavaScript-style comments.
> > +* **package.json** - create an `eslintConfig` property in your `package.json` file and define your configuration there.
> > +* **Deprecated** - use `.eslintrc`, which can be either JSON or YAML.
>
> Because it's the least specific and most fragile. We don't have to remove
> support, but I want to encourage people to use the more specific file
> formats.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/eslint/eslint/pull/4406/files#r44860133>.
>